src/mlpack/core/optimizers/sgdr/snapshot_sgdr_impl.hpp

```diff
@@ -64,16 +61,17 @@ double SnapshotSGDR<DecomposableFunctionType, UpdatePolicyType>::Optimize(
     optimizer.DecayPolicy().StepSize() = optimizer.StepSize();
   }
 
+  optimizer.DecayPolicy().EpochBatches() = function.NumFunctions() /
+      double(batchSize);
+
   // If a user changed the batch size we have to update the restart fraction
   // of the cyclical decay instantiation.
   if (optimizer.BatchSize() != batchSize)
   {
     batchSize = optimizer.BatchSize();
-    optimizer.DecayPolicy().EpochBatches() = function.NumFunctions() /
-        double(batchSize);
```

Comment:
I'm not sure I follow the change here, can you explain why this was needed?  I think it is probably a simple reason I am overlooking.